src/Symfony/Component/Translation/Translator.php

```diff
@@ -491,4 +492,19 @@ protected function assertValidLocale($locale)
             throw new \InvalidArgumentException(sprintf('Invalid "%s" locale.', $locale));
         }
     }
+
+    /**
+     * @see WarmableInterface
+     *
+     * @param string $cacheDir only used for interface compatibility, $this->cacheDir is already defined
+     */
+    public function warmUp($cacheDir)
+    {
+        if ($this->locale !== null) {
```

Comment:
when cacheDir is null we don't need to call initializeCatalogue.
see https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Translation/Translator.php#L324